base/sparse/linalg.jl

```diff
@@ -532,11 +531,14 @@ function normestinv{T}(A::SparseMatrixCSC{T}, t::Integer = 2)
     end
     ind = Array(Int64, n)
     ind_hist = Array(Int64, maxiter * t)
-    S = zeros(T, n, t)
+
+    Ti = typeof(float(zero(T)))
```

Comment:
What is the reasoning for this change? `S` is actually always integer-valued, right, so it could right as well be defined as a an integer matrix.